benchmark.js

```diff
@@ -30,24 +31,60 @@ const CONFIG = {
 };
 
 const outputDir = [__dirname, "datasets", CONFIG.model].join("/");
+var dataDir = [outputDir, CONFIG.split].join("/");
 
 var images = [];
 var buffers = {};
 
 downloadDataset()
 .then(prepareData)
-.then(warmup).then(function() {
-    var begin = Date.now();
-    async.eachOfLimit(buffers, CONFIG.parallelism, function(buffer, path, cb) {
-        infer(buffer, path)
-            .finally(function() {
-                cb(null);
-            });
-    }, function() {
-        var elapsed = ((Date.now() - begin)/1000).toFixed(2);
-        console.log("Inferred", images.length, "times in", elapsed, "seconds", images.length/elapsed, "fps");
+.then(warmup)
+.then(inferAll)
+.then(calculateAccuracy);
+
+function inferAll() {
+    return new Promise(function(resolve, reject) {
+        var begin = Date.now();
+        async.eachOfLimit(buffers, CONFIG.parallelism, function(buffer, path, cb) {
+            infer(buffer, path)
+                .finally(function() {
+                    cb(null);
+                });
+        }, function() {
+            var elapsed = ((Date.now() - begin)/1000).toFixed(2);
+            console.log("Inferred", images.length, "times in", elapsed, "seconds", images.length/elapsed, "fps");
+
+            resolve();
+        });
     });
-});
+}
+
+function calculateAccuracy() {
+    console.log("Calculating Accuracy");
+
+    var formattedPredictions = [];
+    _.each(allPredictions, function(prediction, path) {
+        _.each(prediction, function(box) {
+            formattedPredictions.push({
+                filename: path,
+                confidence: box.confidence,
+                label: box.class,
+                left: box.x - box.width/2,
+                top: box.y - box.height/2,
+                bottom: box.y + box.height/2,
+                right: box.x + box.width/2
```

Comment:
I don't think it'll matter since we're using objects but `right` and `bottom` have been swapped when compared to the docs.

Docs object:
`{
		filename: 'image1.jpg',
		confidence: 0.9,
		label: 'car',
		left: 25,
		top: 38,
		right: 201,
		bottom: 90
}`

Reply:
I’m old 😂  apparently this is not true anymore. But still should hold as not important here https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order